### ISSUE TITLE
False negative with Credo.Check.Readability.ModuleDoc in nested modules

### ISSUE THREAD

**Issue by idkCpp** [rrrene/credo#1168]
### Environment

* Credo version (`mix credo -v`): 1.7.10
* Erlang/Elixir version (`elixir -v`): Elixir 1.17.3 (compiled with Erlang/OTP 27)
* Operating system: Linux

### What were you trying to do?

When having nested module definitions, an inner module having `@moduledoc` will silence the credo check for the outer module.

Consider the following code:

```elixir
defmodule Foo do
  # distinctly no moduledoc here
  defmodule Bar do
    @moduledoc false
  end
end
```

### Expected outcome

```
  Code Readability                                                              
┃ 
┃ [R] → Modules should have a @moduledoc tag.
┃       example.ex:1:11 #(Foo)
```

### Actual outcome

found no issues.

**Comment by rrrene** [comment]
Thx for reporting! :+1: I added a test reproducing this 👍

**Comment by rrrene** [comment]
Thanks for reporting this 😀  It is now fixed on `master`.

You can try this by setting the Credo dep to

```elixir
{:credo, github: "rrrene/credo"}
```

Please report back if your issue is solved! 👍

**Comment by idkCpp** [comment]
I can confirm it to be working.

Many thanks for your time and effort. And thank you for this helpful project!
